src/GraphODataTemplateWriter/CodeHelpers/Java/TypeHelperJava.cs

```diff
@@ -95,27 +89,19 @@ public static string GetToLowerFirstCharName(this OdcmProperty property)
 
         public static string SanitizePropertyName(this string property, OdcmObject odcmProperty = null)
         {
-            if (ReservedNames.Contains(property))
+            if (ReservedNames.Value.Contains(property))
             {
-                logger.Info("Property \"{0}\" is a reserved word in Java. Converting to \"{1}{0}\"", property, ReservedPrefix);
-                return ReservedPrefix + property;
+                var result = ReservedPrefix + property.ToUpperFirstChar();
+                logger.Info($"Property \"{property}\" is a reserved word in Java. Converting to \"{result}\"");
+                return result;
             }
-
-            if (odcmProperty != null && property == odcmProperty.Name.ToUpperFirstChar())
+            else if (property == odcmProperty?.Name?.ToUpperFirstChar() && !(odcmProperty?.Name?.StartsWith("_") ?? false))
```

Comment:
no need for `else`

Reply:
it's a preference thing. I know it's a bit more verbose but it eases up my reading as I don't need to go into the body of the previous branch to understand only one of the branches will execute.